File: app/src/main/java/com/nollpointer/dates/other/Date.java
package com.nollpointer.dates.other;

import android.os.Parcel;
import android.os.Parcelable;

public class Date implements Parcelable, Comparable<Date>{
    private String date;
    private String event;
    private String request;
    private int type;

    private String month;

    public Date(String date, String event, String request, int type) {
        this.date = date;
        this.event = event;
        this.request = request;
        this.type = type;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getEvent() {
        return event;
    }

    public void setEvent(String event) {
        this.event = event;
    }

    public String getRequest() {
        return request;
    }

    public void setRequest(String request) {
        this.request = request;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getMonth() {
        return month;
    }

    public void setMonth(String month) {
        this.month = month;
    }

    public boolean containsMonth(){
        return month != null;
    }

    protected Date(Parcel in) {
        date = in.readString();
        event = in.readString();
        request = in.readString();
        type = in.readInt();
        month = in.readString();
    }

    public boolean isSameDate(Date date){
        return this.getDate().equals(date.getDate());
    }

    public boolean isContinuous (){
        return date.contains("-") || date.contains("–") || date.contains("–") || date.contains(",") || date.contains("в");
    }

    public boolean contains(String query){
        return date.toLowerCase().contains(query.toLowerCase()) || event.toLowerCase().contains(query.toLowerCase());
    }

    @Override
    public int compareTo(Date o) {
        int currentDate = Integer.parseInt(date);
        int compareDate = Integer.parseInt(o.date);

        return Integer.compare(currentDate,compareDate);
    }

    //Parcelable implementation

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int flags) {
        parcel.writeString(date);
        parcel.writeString(event);
        parcel.writeString(request);
        parcel.writeInt(type);
        parcel.writeString(month);
    }

    public static final Parcelable.Creator<Date> CREATOR = new Parcelable.Creator<Date>() {
        @Override
        public Date createFromParcel(Parcel in) {
            return new Date(in);
        }

        @Override
        public Date[] newArray(int size) {
            return new Date[size];
        }
    };

}

File: app/src/main/java/com/nollpointer/dates/other/Date.kt
package com.nollpointer.dates.other

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable
import kotlinx.android.parcel.Parceler
import kotlinx.android.parcel.Parcelize
import java.util.*


@Parcelize
data class Date(var date: String,
                var event: String,
                var request: String,
                var type: Int,
                var month: String? = null) : Parcelable, Comparable<Date> {

    val isContinuous: Boolean
        get() = date.contains("-") || date.contains("–") || date.contains("–") || date.contains(",") || date.contains("в")

    fun containsMonth(): Boolean {
        return month != null
    }

    fun isSameDate(date: Date): Boolean {
        return this.date == date.date
    }


    operator fun contains(q: String): Boolean {
        val query = q.toLowerCase(Locale.ROOT)
        return date.toLowerCase(Locale.ROOT).contains(query) || event.toLowerCase(Locale.ROOT).contains(query) || containsMonth() && month!!.toLowerCase(Locale.ROOT).contains(query)
    }

    override fun compareTo(other: Date): Int {
        val currentDate = date.toInt()
        val compareDate = other.date.toInt()
        return currentDate.compareTo(compareDate)
    }
}